bug: Panic in color.Trim()

Hi, I'm suffering this panic with completions:

```
panic: runtime error: slice bounds out of range [:303] with length 302

goroutine 1 [running]:
github.com/reeflective/readline/internal/color.Trim({0x14001102640, 0x12e}, 0xde)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/color/color.go:113 +0xd4
github.com/reeflective/readline/internal/completion.(*group).trimDesc(0x140006984e0, {{0x140006b4470, 0xd}, {0x140006b4d10, 0x10}, {0x1400112c1c1, 0x138}, {0x0, 0x0}, {0x1022d3ccd, ...}, ...}, ...)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/completion/group.go:387 +0x108
github.com/reeflective/readline/internal/completion.(*Engine).highlightDesc(0x14001188000, 0x140006984e0, {{0x140006b4470, 0xd}, {0x140006b4d10, 0x10}, {0x1400112c1c1, 0x138}, {0x0, 0x0}, ...}, ...)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/completion/display.go:149 +0x84
github.com/reeflective/readline/internal/completion.(*Engine).renderCompletions(0x14001188000, 0x140006984e0)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/completion/display.go:93 +0x680
github.com/reeflective/readline/internal/completion.Display(0x14001188000, 0xc)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/completion/display.go:34 +0x110
github.com/reeflective/readline/internal/display.(*Engine).displayHelpers(0x140001a6210)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/display/engine.go:306 +0x98
github.com/reeflective/readline/internal/display.(*Engine).Refresh(0x140001a6210)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/internal/display/engine.go:95 +0x104
github.com/reeflective/readline.(*Shell).Readline(0x14000240640)
        /Users/liamg/go/pkg/mod/github.com/reeflective/readline@v1.1.3/readline.go:84 +0x174
github.com/liamg/shiv/internal/shell.(*Shell).StartInteractive(0x140002403c0)
        /Users/liamg/src/shiv/internal/shell/shell.go:270 +0x4f4
github.com/liamg/shiv/internal/command.init.func2(0x1400022e700?, {0x14000028950, 0x0, 0x1})
        /Users/liamg/src/shiv/internal/command/root.go:109 +0x364
github.com/spf13/cobra.(*Command).execute(0x102886a40, {0x140000120b0, 0x1, 0x1})
        /Users/liamg/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x102886a40)
        /Users/liamg/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/liamg/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /Users/liamg/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1064
github.com/liamg/shiv/internal/command.ExecuteContext(...)
        /Users/liamg/src/shiv/internal/command/root.go:19
main.main()
        /Users/liamg/src/shiv/cmd/shiv/main.go:13 +0x58
```

I managed to dump the input causing this to use as a test case:
```
"\x1b[34m1\x1b[0m      \x1b[33mname       \x1b[0m  env    \x1b[33mdescription\x1b[0m  Read and write environment variables    \x1b[34m2\x1b[0m      \x1b[33mname       \x1b[0m  script    \x1b[33mdescription\x1b[0m  Script info, such as cmd and args    \x1b[34m3\x1b[0m      \x1b[33mname       \x1b[0m  system    \x1b[33mdescription\x1b[0m  System information"
```

I might have some time to debug further and PR a fix around this time tomorrow.

Note: The input above looks weird because I called the wrong stringer - but the bug stands nonetheless 😅

--- a/internal/color/color.go
+++ b/internal/color/color.go
@@ -106,6 +106,10 @@ func Trim(input string, maxPrintableLength int) string {
 		}
 	}
 
+	if maxPrintableLength > len(input) {
+		return input
+	}
+
 	// Determine the end index for limiting printable content
 	return input[:maxPrintableLength]
 }

